proxy/server_quic.go

```diff
@@ -137,6 +137,19 @@ func (p *Proxy) handleQUICStream(stream quic.Stream, session quic.Session) {
 		log.Info("failed to unpack a DNS query: %v", err)
 	}
 
+	// If any message sent on a DoQ connection contains an edns-tcp-keepalive EDNS(0) Option,
```

Comment:
Is there a link to the spec that explains this?

Reply:
Sure, quoting [draft-ietf-dprive-dnsoquic-02 section 6.6.2](https://tools.ietf.org/html/draft-ietf-dprive-dnsoquic-02#section-6.6.2):

> Clients that want to maintain long duration DoQ connections SHOULD
> use the idle timeout mechanisms defined in Section 10.2 of the QUIC
> transport specification [I-D.ietf-quic-transport].  Clients and
> servers MUST NOT send the edns-tcp-keepalive EDNS(0) Option [RFC7828]
> in any messages sent on a DoQ connection (because it is specific to
> the use of TCP/TLS as a transport).  If any message sent on a DoQ
> connection contains an edns-tcp-keepalive EDNS(0) Option, this is a
> fatal error and the recipient of the defective message MUST forcibly
> abort the connection immediately.

Reply:
I mean it'd be better to link it in the code :)

Reply:
Ah yes, that makes sense :) Added in 059cdd3